base/show.jl

```diff
@@ -12,6 +12,13 @@ end
 
 show(io::IO, ::MIME"text/plain", r::AbstractRange) = show(io, r) # always use the compact form for printing ranges
 
+function show(io::IO, ::MIME"text/plain", r::UnitRange)
+    print(io, repr(first(r)), ':', repr(last(r)))
```

Comment:
More efficient:

```suggestion
    show(io, first(r))
    print(io, ':')
    show(io, last(r))
```

Proof:

```julia-repl
julia> f(io, a, b) = print(io, repr(a), ':', repr(b))
f (generic function with 1 method)

julia> function g(io, a, b)
           show(io, a)
           print(io, ':')
           show(io, b)
       end
g (generic function with 1 method)

julia> using BenchmarkTools

julia> @btime f(io, a, b) setup=(io = devnull; a = big(2)^(2^10); b = a + 1;)
  2.547 μs (12 allocations: 1.62 KiB)

julia> @btime g(io, a, b) setup=(io = devnull; a = big(2)^(2^10); b = a + 1;)
  2.201 μs (4 allocations: 736 bytes)
```